### ISSUE TITLE
Provide example code

### ISSUE THREAD

**Issue by Goldenbrain1** [andreasdr/tdme2#11]
I have never used any 3D engine before, and I'm a bit lost, i think providing examples of programs using the engine would help people like me to understand how the engine works.

**Comment by andreasdr** [comment]
Hi,

Many thanx for your interest.

TDME2 is maybe not yet ready for just playing around with easily.

We have lots of plans, including adding lots of usability fixes, after releasing our "A Scottish War Story" game alpha in a few months.

We will also add a script language, most likely "Haxe" and add support in the tools for it. 
Then you can script games in a single tool.

Anyways I will need to add lots of documentation soon like
- public API of all modules
- UI system
- ...
- and maybe something like best practices

As TDME2 is more of a visualization engine it lacks some features that are included in a game engine like UE.

However our ASW has some sort of basement that you can think of a game engine.
It can happen that we will open source this too.

Some examples for now live here:
- https://github.com/andreasdr/tdme2/tree/master/src/tdme/tests

So maybe its the best to wait a little bit. However game development requires a bit of math knowledge. I could also write some documentation about this topic.

There is lots of stuff to do still.

Many thanx and

Best regards
Andreas

**Comment by Goldenbrain1** [comment]
Oh thank you, well i just needed a bit of help on how to begin a program using the engine, I will already try like that then learn how to use all the features when the documentation is finished, it seems that there is already quite a bit of features.
The programs that you linked will help me to begin i think, shouldn't you paste that link in the readme ?
For math knowledge, don't worry I don't think a program ever made me math difficulties or will be (unless i try to build a non-euclidian spherical render engine lol)

**Comment by andreasdr** [comment]
Yes! Nice. Have a look. If you have any suggestions or anything else, just let me know.
I also have a Demonstration Video and Screenshots in the making. But it will take some time still.

**Comment by niansa** [comment]
I agree, examples are important.
